apps/src/code-studio/pd/application/facilitator1819/Facilitator1819FormComponent.jsx

```diff
@@ -1,6 +1,11 @@
-import {PropTypes} from 'react';
+import React, {PropTypes} from 'react';
+import marked from 'marked';
 import FormComponent from '../../form_components/FormComponent';
 
+// Render links to open in a different tab
+const renderer = new marked.Renderer();
+renderer.link = (href, _title, text) => `<a href="${href}" target="_blank">${text}</a>`;
+
 export default class Facilitator1819FormComponent extends FormComponent {
```

Comment:
Inheritance is generally discouraged when using React (composition is preferred). It looks like `FormComponent` is used pretty widely at this point, but worth considering as you do future designs.

Reply:
@Hamms built this [form infrastructure](https://github.com/code-dot-org/code-dot-org/tree/staging/apps/src/code-studio/pd/form_components) to replace Pegasus forms for all these dashboard pd forms and applications. Is inheritance always discouraged in React? I'm not entirely sure what his design motivation was, but to me this seems like a good candidate for inheritance. The form pages and components themselves use composition but they all derive from base types that provide functionality. Here I was following that trend, but I could switch it to a wrapper component that passes props through if you think that's better.

Reply:
I think it's almost always discouraged. Here's an article I found on it. https://reactjs.org/docs/composition-vs-inheritance.html

I think if this is working well for you guys, it's fine to stick with it, but would recommend avoiding adding much more inheritance if you can.

Reply:
I think I read that exact same article when I was trying to decide how to build these.

I went with inheritance over strict composition because I wanted to leverage class methods when defining the relationship between FormController and its various FormComponents; I did initially attempt to do what they recommend and extract the functionality into a non-React class, but it ended up being super cumbersome.

I do feel like this pattern has been working well so far, but I'm open to contradictory feedback